icuSources/CMakeLists.txt

```diff
@@ -0,0 +1,49 @@
+##===----------------------------------------------------------------------===##
+##
+## This source file is part of the SwiftFoundation open source project
+##
+## Copyright (c) 2024 Apple Inc. and the SwiftFoundation project authors
+## Licensed under Apache License v2.0
+##
+## See LICENSE.txt for license information
+## See CONTRIBUTORS.md for the list of SwiftFoundation project authors
+##
+## SPDX-License-Identifier: Apache-2.0
+##
+##===----------------------------------------------------------------------===##
+
+add_library(_FoundationICU)
+
+target_include_directories(_FoundationICU
+    PRIVATE
```

Comment:
Since we're adding each subdirectory as an include directory with `target_include_directories(_FoundationICU PRIVATE .)`, we shouldn't need any of these `PRIVATE` include directories here either.